File: Lista_Logica_Programacao/Java/Ex7.java
// Exercício 7 - Maior, menor e média de uma lista de números

import java.util.Scanner;

public class Ex7 {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("Quantos números? ");
        int quantidade = sc.nextInt();

        double[] numeros = new double[quantidade];

        for (int i = 0; i < quantidade; i++) {
            numeros[i] = sc.nextDouble();
        }

        double maior = numeros[0];
        double menor = numeros[0];
        double soma = 0;

        for (int i = 0; i < quantidade; i++) {
            if (numeros[i] > maior) {
                maior = numeros[i];
            }
            if (numeros[i] < menor) {
                menor = numeros[i];
            }
            soma += numeros[i];
        }

        double media = soma / quantidade;

        System.out.println("Maior: " + maior);
        System.out.println("Menor: " + menor);
        System.out.println("Média: " + media);

        sc.close();
    }
}
/*
Anotações:

- Usei array porque preciso armazenar vários valores
- double[] numeros cria vetor com tamanho definido
- Precisei iniciar maior e menor com o primeiro valor do array
- Fiz um segundo for para comparar manualmente
- Java não tem max() e min() prontos para array simples
- Precisei calcular soma manualmente
*/

File: Lista_Logica_Programacao/Python/ex7.py
# Exercício 7 - Lista de números

quantidade = int(input("Quantos números você quer digitar? "))

numeros = []

for i in range(quantidade):
    num = float(input("Digite um número: "))
    numeros.append(num)  # adiciona na lista

maior = max(numeros)
menor = min(numeros)
media = sum(numeros) / len(numeros)

print("Maior:", maior)
print("Menor:", menor)
print("Média:", media)

# Anotações:
# - Usei lista porque a quantidade é variável
# - append() adiciona elemento na lista
# - max() e min() já existem prontos em Python
# - sum() soma todos os elementos
# - len() retorna o tamanho da lista
# - Python tem muitas funções prontas que facilitam